protocols/autonatv2/src/client/handler/dial_back.rs

```diff
@@ -0,0 +1,93 @@
+use std::{
+    convert::identity,
+    io,
+    task::{Context, Poll},
+};
+
+use either::Either;
+use futures::{AsyncRead, AsyncWrite, AsyncWriteExt};
+use futures_bounded::{FuturesSet, Timeout};
+use libp2p_core::upgrade::{DeniedUpgrade, ReadyUpgrade};
+use libp2p_swarm::{
+    handler::{ConnectionEvent, FullyNegotiatedInbound, ListenUpgradeError},
+    ConnectionHandler, ConnectionHandlerEvent, StreamProtocol, SubstreamProtocol,
+};
+
+use crate::request_response::DialBack;
+
+use super::DEFAULT_TIMEOUT;
+
+pub(crate) type ToBehaviour = Result<u64, io::Error>;
+
+pub(crate) struct Handler {
+    inbound: FuturesSet<io::Result<u64>>,
+}
+
+impl Handler {
+    pub(crate) fn new() -> Self {
+        Self {
+            inbound: FuturesSet::new(DEFAULT_TIMEOUT, 2),
+        }
+    }
+}
+
+impl ConnectionHandler for Handler {
+    type FromBehaviour = ();
+    type ToBehaviour = ToBehaviour;
+    type InboundProtocol = ReadyUpgrade<StreamProtocol>;
+    type OutboundProtocol = DeniedUpgrade;
+    type InboundOpenInfo = ();
+    type OutboundOpenInfo = ();
+
+    fn listen_protocol(&self) -> SubstreamProtocol<Self::InboundProtocol, Self::InboundOpenInfo> {
+        SubstreamProtocol::new(crate::DIAL_BACK_UPGRADE, ())
+    }
+
+    fn poll(
+        &mut self,
+        cx: &mut Context<'_>,
+    ) -> Poll<
+        ConnectionHandlerEvent<Self::OutboundProtocol, Self::OutboundOpenInfo, Self::ToBehaviour>,
+    > {
+        if let Poll::Ready(result) = self.inbound.poll_unpin(cx) {
+            return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(
+                result
+                    .map_err(|timeout| io::Error::new(io::ErrorKind::TimedOut, timeout))
+                    .and_then(identity),
```

Comment:
Mild preference to replace the `if let` with a `match` and simply extend the pattern to match on `Poll::Ready(Ok(..))` and `Poll::Ready(Err(..))`.

Reply:
That is changed.